picaso/justdoit.py

```diff
@@ -2713,34 +2714,130 @@ def atmosphere_4d(self, ds=None, shift=None, plot=True, iz_plot=0,verbose=True,
         else :
             og_lat = ds.coords['lat'].values #degrees
             og_lon = ds.coords['lon'].values #degrees
-
+            pres = ds.coords['pressure'].values #bars  #adding this so I can add it explicitly to ds_New below
+            
         #store so we can rotate
         data_vars_og = {i:ds[i].values for i in ds.keys()}
         #run through phases and regrid each one
+        new_lat_totals = []
+        new_lon_totals = []
+        #new_lat_totals_og = []
+        new_lon_totals_og = []
         shifted_grids = {}
+
+        # Add if calculation = reflected here:
```

Comment:
The `clouds_4d` has this if statement for reflected and thermal. Does atmosphere_4d need it as well?

Reply:
Atmosphere_4d didn't end up needing an if statement here. The conditional statements in Line 2928 - 2930 (and Lines 2951 - 2953) make it so that the thermal phase curve functionality is not affected, but the parts of the reflected case that are not at full phase are shifted as needed.